Ab/deforest

Alucowie opened the pull request:
<!-- Please read the contributing guidelines:
* https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md
In short:
* PR title must be capitalized, concise and use ##tags
* Follow the coding style, add tests and documentation if necessary
-->

**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

A WIP attempt to fix #18361

radare committed:
Remove tree-sitter and the r2-shell-parser ##shell

* Will be reimplemented soon with a new parser
* Simplifies and speedup the build

Alucowie committed:
Fix type "intgerp" -> "intgrep"

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
cmd: flush console buffer between multi-command

Tests fixed:
db/cmd/feat_grep multi-command single-line grep
db/cmd/cmd_zignature af ; af ; zfs libc-v10.sig - newshell
db/cmd/cmd_zignature af ; af ; zfs libc-v7.sig - newshell

New test failed:
db/tools/rax2 C array {0x33,33,23,32}

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
cmd: double quotes escape the semi-colon command separator

Simple quotes already has.

Test fixed:
db/tools/rax2 C array {0x33,33,23,32}

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
cmd: simple quotes also escape the hash comment

Add a new test:
db/cmd/cmds hash comment and simple quote

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
unescape the alias' argument

Test fixed:
db/cmd/cmd_interpret $foo=#!pipe

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
Remove refs to radare2-shell-parser

This fix the CI build of linux-static.

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
debug: Do not print 0 on unknown register

Fix CI linux-test.

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
Trim arguments to be closer to echo

Test fixed:
db/cmd/cmds hash comment and simple quote

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
simple and double quotes also unescape.

Partially fix:
db/cmd/feat_quote Single quotes - newshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

trufae commented on libr/core/cmd.c:
```diff
@@ -3174,6 +3174,9 @@ static int r_core_cmd_subst(RCore *core, char *cmd) {
 			break;
 		}
+		if (colon) {
+			r_cons_flush();
```
uhm that may break r2pipe and missing space

trufae commented on libr/core/cmd.c:
```diff
@@ -702,6 +702,7 @@ static int cmd_alias(void *data, const char *input) {
 			def[len - 1] = 0x00;
 			def++;
 		}
+		r_str_arg_unescape(def);
```
missing space

Alucowie committed:
Write cmd also need to handle quotes like echo cmd

Partially fix:
db/cmd/feat_quote Double quotes - newshell
db/cmd/feat_quote Single quotes - newshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
Protect arroba under quotes.

Tests fix:
db/cmd/feat_quote Single quotes - newshell
db/cmd/feat_quote Double quotes - newshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
Do not execute subcommand under simple quotes protection.

Partially fix:
db/cmd/feat_quote Quotes and substitution - newshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
backtick is a known escape subcommand character.

New test in:
db/cmd/feat_quote Quotes and substitution - newshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
Handle $() and `` as equal sub-command.

Test fix:
db/cmd/feat_quote Quotes and substitution - newshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
Trim tail sub-command buffer to avoid tailing space

Finally fix:
db/cmd/feat_quote Quotes and substitution - newshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
Mark oldshell as broken

Broken tests:
db/cmd/regexp - "/e /t\wst\d\d\d\s\w\w/i" - oldshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>

Alucowie committed:
Test "db/cmd/cmd_seek seekinstr" is valid with newshell

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@gmail.com>
